### PR TITLE
Introduce go modules [yey]

### PR DESCRIPTION
**Pull request by Kubuxu**
Makefiles and CI were adapted.

socat commit: https://github.com/ipfs/go-ipfs/pull/6038/commits/f9392ed61c57c1649b848c8007493b53e03aa675

### CODE REVIEW

**Comment by Kubuxu** [comment]
`gotest` is broken because our JUnit test results generation is broken in general. Waiting for @magik6k to fix it.

**Comment by Kubuxu** [comment]
I have killed all uses of `netcat` inside sharness tests because differences between bsd and gnu netcat were making me mad. I've replaced it with `socat` which is more complex but better suited for the job in most places.

**Comment by Kubuxu** [comment]
@Stebalien this PR completly removed `gx` from Makefiles. It was much easier to do than to try to negotate gomod with gx.

Is this OK solution? How do yo feel about it. What would be the next step to getting this merged?

**Comment by Kubuxu** [comment]
A more useful commit range: https://github.com/ipfs/go-ipfs/pull/6038/files/a1b0c0309d0d95e4386eba58658825b487918573..68c1035ad4641a58332484d5ff9b5f73c4a63a76

**Comment by Kubuxu** [comment]
I've fixed JUnit for gotest generation and cleaned the circleci config a bit.

**Comment by Stebalien** [comment]
Is the socat change going to cause trouble for macos users?

**Comment by Kubuxu** [comment]
`socat` is available on https://brewinstall.org/Install-socat-on-Mac-with-Brew/

**Comment by Stebalien** [comment]
(will merge once @magik6k signs off on it)

File: test/unit/Rules.mk [hunk 1]
```diff
@@ -2,7 +2,7 @@ include mk/header.mk
 
 CLEAN += $(d)/gotest.json $(d)/gotest.junit.xml
 
-$(d)/gotest.junit.xml: clean test/bin/json-to-junit coverage/unit_tests.coverprofile
-	cat $(@D)/gotest.json | json-to-junit > $(@D)/gotest.junit.xml
```
Comment (magik6k) [inline test/unit/Rules.mk]: We should remove `test/dependencies/json-to-junit`

**Comment by Stebalien** [comment]
:rocket:

**Comment by Kubuxu** [comment]
Yey. Put it on a boat.

**Pull request merged**